aws/iam_test.go

```diff
@@ -0,0 +1,51 @@
+package aws
+
+import (
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/iam"
+
+	"github.com/gruntwork-io/cloud-nuke/config"
+	"github.com/gruntwork-io/gruntwork-cli/errors"
+	"github.com/stretchr/testify/assert"
+)
+
+//TODO: This test should check that IAM Roles with path /aws-service-role/
+// are not part of the nukable roles, and fail if so.
+func TestListIamRoles(t *testing.T) {
+	sess, err := session.NewSession(&aws.Config{})
+
+	input := &iam.ListRolesInput{}
+
+	_, err = getAllIamRoles(sess, input)
+	if err != nil {
+		assert.Fail(t, "Failed to list IAM Roles", errors.WithStackTrace(err))
+	}
+}
+
+func TestNukeIamRoles(t *testing.T) {
+	sess, err := session.NewSession(&aws.Config{})
+
```

Comment:
Flow looks good! One thing is that this test will need to be updated so that it is not as heavily dependent on the environment. (E.g., creating a test IAM role in the test)